nilearn/regions/parcellations.py

```diff
@@ -40,26 +43,18 @@ def _estimator_fit(data, estimator, method=None):
         labels_ estimated from estimator.
 
     """
-    if method == "rena":
-        rena = ReNA(
-            mask_img=estimator.mask_img,
-            n_clusters=estimator.n_clusters,
-            scaling=estimator.scaling,
-            n_iter=estimator.n_iter,
-            threshold=estimator.threshold,
-            memory=estimator.memory,
-            memory_level=estimator.memory_level,
-            verbose=estimator.verbose,
-        )
-        rena.fit(data)
-        labels_ = rena.labels_
-
+    estimator = clone(estimator)
+    if method in ["rena", "hierarchical_kmeans"]:
+        estimator.fit(data)
+    # transpose data for KMeans, AgglomerativeClustering because
+    # they cluster first dimension of data (samples) but we want to cluster
+    # features (voxels)
     else:
-        estimator = clone(estimator)
         estimator.fit(data.T)
```

Comment:
By "issue" I am just referring to the transpose of samples and features between nilearn vs. sklearn.

Since we have methods like kmeans and wards that use sklearn implementations we do this explicit transpose of the data for those methods here and no transpose for ReNA and now HierarchicalKMeans too

Reply:
I see, so this is still the same point about aligning the user interface between parcellation methods, it's just that in some parcellation cases we were handling this internally (i.e., imposing the idea that users would want to find clusters in voxels rather than in time points) while in ReNA and HierarchicalKMeans we were allowing the user to pass through directly to the sklearn API, which would have given unexpected results (assuming that they indeed want clusters in voxels, which almost all neuroimagers do with these methods) unless they transposed the data themselves.

Is that an accurate summary ? If so, I'd say that we're indeed just streamlining user experience 😸  In a very important way !

Reply:
Yes indeed! It isn't a problem with our pipeline or method